Remotion/Web/UnitTests/Core/Resources/ResourcePathBuilderBasedStaticResourceCacheKeyProviderTest.cs

```diff
@@ -0,0 +1,310 @@
+
+      folder.CreateFile(@"Remotion.Web\Html\test.txt", "blabla", DateTime.FromFileTime(130000011111111111), DateTime.FromFileTime(130000022222222222));
+
+      WaitForCacheKeyChange(testStaticResourceCacheKeyProvider);
+
+      var expectedCacheKey = CreateExpectedCacheKeyForEntries(
+          (@"Remotion.Web\Html\test.txt", 6, 130000011111111111, 130000022222222222));
+      Assert.That(testStaticResourceCacheKeyProvider.GetStaticResourceCacheKey(), Is.EqualTo(expectedCacheKey));
+    }
+
+    [Test]
+    public void FileSystemWatcher_WithIrrelevantFileChanges_DoesNotUpdateCacheKey ()
+    {
+      using var folder = TempFolder.Create();
+
+      var resourceTypes = new[] { ResourceType.Html };
+
+      var resourcesPhysicalPathLocatorMock = new Mock<ResourcePathBuilderBasedStaticResourceCacheKeyProvider.IMappedResourcesPathLocator>();
+      resourcesPhysicalPathLocatorMock.Setup(_ => _.GetMappedResourcesPath()).Returns(folder.FullName);
+
+      using var testStaticResourceCacheKeyProvider = new TestStaticResourceCacheKeyProvider(
+          resourcesPhysicalPathLocatorMock.Object,
+          new ResourcePathBuilderBasedStaticResourceCacheKeyProvider.ResourceFileDetailsAppender(),
+          resourceTypes);
+
+      Assert.That(testStaticResourceCacheKeyProvider.GetStaticResourceCacheKey(), Is.EqualTo(CreateExpectedCacheKeyForEntries()));
+
+      // Changes in the root folder should not be considered as long as they aren't deletions or renames
+      folder.CreateDirectory("Remotion.Web");
+      folder.CreateFile("abc.txt", "testContent", DateTime.Today, DateTime.Today);
+
+      // Changes to the assembly folders should not be considered as long as they aren't deletions or renames
+      folder.CreateDirectory(@"Remotion.Web\Html");
+      folder.CreateDirectory(@"Remotion.Web\UI");
+      folder.CreateFile(@"Remotion.Web\abc.txt", "testContent", DateTime.Today, DateTime.Today);
+
+      // Changes to relevant folders only consider files not directories
+      folder.CreateDirectory(@"Remotion.Web\Html\subfolder", DateTime.Today, DateTime.Today);
+
+      // Changes to irrelevant folders should be ignored completely
+      folder.CreateDirectory(@"Remotion.Web\UI\subfolder");
+      folder.CreateFile(@"Remotion.Web\UI\subfolder\abc.txt", "testContent", DateTime.Today, DateTime.Today);
+
+      // No better way than to sleep as we expect no notifications to come in and
+      // they can have an arbitrary delay. 100ms should be good enough as tests
+      // with 50ms also worked well.
+      Thread.Sleep(100);
+
+      Assert.That(testStaticResourceCacheKeyProvider.HasCacheKey, Is.True);
+    }
+
+    private void WaitForCacheKeyChange (TestStaticResourceCacheKeyProvider staticResourceCacheKeyProvider)
+    {
+      for (var i = 0; i < 10; i++)
+      {
+        if (!staticResourceCacheKeyProvider.HasCacheKey)
+          return;
+
+        Thread.Sleep(20);
```

Comment:
api note: I like to use Timespan.FromMillisconds instead of presuming the reader knows the API. It's not somehting I need to change, it's just somehting for future consideration.

Reply:
![image](https://github.com/re-motion/Framework/assets/5082673/c5710da3-e728-4b59-ab5f-59af247dee57)
;)

Reply:
![image](https://github.com/re-motion/Framework/assets/6972548/380e41c5-cbc7-4bc7-be83-20dc7c81205a)

:p

Code should always be readable without IDE hints:
There's too many scenarios where you don't have Rider/Resharper enabled (Teams that don't use those tools, decompiled code, viewed on the website of a source control system or a depedency analyzer. And devs who don't like the inline hits because it messes with alignment and line length experience).